pkg/vsphere/vm/vm.go

```diff
@@ -87,17 +110,41 @@ func (vm VirtualMachine) WaitForMAC(ctx context.Context) (map[string]string, err
 					if mac == "" {
 						return false
 					}
-					summary := nic.GetVirtualEthernetCard().DeviceInfo.GetDescription().Summary
-					macs[summary] = mac
+					netName := nicMappings[nic.GetVirtualEthernetCard().DeviceInfo.GetDescription().Summary]
```

Comment:
don't return false in this loop if mac == "" - that means we will not gather the values from other devices reported in this propertychange set. `Return false` suffices for a binary done/not done test but not for building a mapping.